api/renewer.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func init() {
+	// Seed the random generator
+	rand.Seed(time.Now().Unix())
```

Comment:
I thought about this, and I really don't have a good answer. If consumers of the client library don't randomize, we completely lose the point of randomization, since they'll randomize on the same randomness. I'm not opposed to removing or keeping it. If there's a pre-defined pattern, I'd love to learn more about it.